src/wp-includes/script-loader.php

```diff
@@ -279,6 +279,10 @@ function wp_default_packages_scripts( $scripts ) {
 	 */
 	$assets = include ABSPATH . WPINC . "/assets/script-loader-packages{$suffix}.php";
 
+	// Add the private version of the Interactivity API manually.
+	$scripts->add( 'wp-interactivity', '/wp-includes/js/dist/interactivity.min.js' );
+	did_action( 'init' ) && $scripts->add_data( 'wp-interactivity', 'strategy', 'defer' );
```

Comment:
Per my response in https://github.com/WordPress/wordpress-develop/pull/5262#discussion_r1337219123, given we want to avoid exposing this API too much until it's stable, how about we don't register it unconditionally?

We could introduce a function like this:
```php
function _wp_maybe_enqueue_interactivity() {
	if ( wp_script_is( 'wp-interactivity' ) ) {
		return;
	}
	wp_enqueue_script(
		'wp-interactivity',
		'/wp-includes/js/dist/interactivity.min.js',
		array(),
		false,
		array( 'strategy' => 'defer' )
	);
}
```

That function could then be called in all the blocks where it is needed, and with such an approach the use of the `wp-interactivity` script would be more obviously discouraged.

Reply:
cc. @luisherranz for your thoughts here.

Reply:
> given we want to avoid exposing this API too much until it's stable, how about we don't register it unconditionally?

1. If your worries are about third-party scripts accessing the exports that are inside of that JavaScript file:

    That file is generated by Webpack as [an independent _chunk_](https://github.com/WordPress/wordpress-develop/blob/e3a612a97c4c45acd7613779a46d86860e76cc65/tools/webpack/blocks.js#L92-L98). Only [the entry points generated in the same webpack config](https://github.com/WordPress/wordpress-develop/blob/e3a612a97c4c45acd7613779a46d86860e76cc65/tools/webpack/blocks.js#L67-L72) have the correct auto-generated ID to access those exports.

    I'm happy to explore other additional measures. For example, I also tested [a random ID that changes each time Webpack runs](https://github.com/luisherranz/wordpress-develop/blob/daca7f4902da198adf3cbae925c69eb3ae78eb52/tools/webpack/blocks.js#L78), although it didn't feel necessary to me as accessing a private Webpack generated chunk is already very hacky and there's no guarantee that the auto-generated ID won't change either.

2. If your worries are about having a script that sometimes is registered unnecessarily and you would prefer to make sure that it's only registered if it's finally enqueued: sure, we can try to add logic to check for that 🙂

Just let me know and I'll take care of it as soon as possible.

Reply:
Related to the random ID:

- https://github.com/WordPress/wordpress-develop/pull/5262#discussion_r1342364649